### PR TITLE
Sponsor tx

### PR DESCRIPTION
**Pull request by diepmv**

### CODE REVIEW

File: core/state_transition.go [hunk 1]
```diff
@@ -530,13 +582,22 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 		// Increment the nonce for the next transaction
 		st.state.SetNonce(msg.From(), st.state.GetNonce(sender.Address())+1)
 
+		sponsorTx, err := st.decodeSponsorTx()
```
Comment (Zergity) [inline core/state_transition.go:585]: just call `decodeSponsorTx` inside `decodeFeePayer` to make this part more clean

**Review by Zergity — CHANGES_REQUESTED** [review]

File: core/state_transition.go (hunk 1 above)
Comment (diepmv) [reply core/state_transition.go]: @Zergity I've pushed a new commit to fix this

File: core/state_transition.go [hunk 2]
```diff
@@ -536,7 +598,6 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 		}
 
 		var savedGas uint64
-
```
Comment (Zergity) [inline core/state_transition.go:539]: please remove unnecessary white-space change

**Review by Zergity — APPROVED** [review]
1 minor refactor and it's good

**Pull request merged**